src/modules/frost/main_impl.h

```diff
@@ -152,67 +156,56 @@ static SECP256K1_WARN_UNUSED_RESULT int initialize_random_scalar(secp256k1_scala
-                                     const unsigned char *msg, uint32_t msg_len, unsigned char *hash_value) {
-    uint32_t ext_msg_len = prefix_len + msg_len;
-    unsigned char *ext_msg = (unsigned char *) checked_malloc(&default_error_callback, ext_msg_len);
-    memcpy(ext_msg, prefix, prefix_len);
-    memcpy(ext_msg + prefix_len, msg, msg_len);
-    compute_sha256(ext_msg, ext_msg_len, hash_value);
-    if (ext_msg != NULL) {
-        free(ext_msg);
-    }
-}
-
 static void compute_hash_h1(const unsigned char *msg, uint32_t msg_len, unsigned char *hash_value) {
     /* TODO: replace with hash-to-curve
     * H1(m): Implemented using hash_to_field from [HASH-TO-CURVE], Section 5.3 using L = 48,
     * expand_message_xmd with SHA-256, DST = contextString || "rho", and prime modulus equal to Order(). */
-    unsigned char prefix[26 + 3];
-    memcpy(prefix, hash_context_prefix, 26);
-    memcpy(prefix + 26, "rho", 3);
-    compute_hash_with_prefix(prefix, sizeof(prefix), msg, msg_len, hash_value);
+    secp256k1_sha256 sha;
+    secp256k1_sha256_initialize(&sha);
+    secp256k1_sha256_write(&sha, hash_context_prefix_h1, sizeof(hash_context_prefix_h1));
+    secp256k1_sha256_write(&sha, msg, msg_len);
+    secp256k1_sha256_finalize(&sha, hash_value);
 }
 
 static void compute_hash_h2(const unsigned char *msg, uint32_t msg_len, unsigned char *hash_value) {
     /* TODO: replace with hash-to-curve
     * H2(m): Implemented using hash_to_field from [HASH-TO-CURVE], Section 5.2 using L = 48,
     * expand_message_xmd with SHA-256, DST = contextString || "chal", and prime modulus equal to Order().*/
     const unsigned char prefix[17] = "BIP0340/challenge";
-    compute_hash_with_prefix(prefix, sizeof(prefix), msg, msg_len, hash_value);
+    secp256k1_sha256 sha;
+    secp256k1_sha256_initialize(&sha);
+    secp256k1_sha256_write(&sha, prefix, sizeof(prefix));
+    secp256k1_sha256_write(&sha, msg, msg_len);
+    secp256k1_sha256_finalize(&sha, hash_value);
 }
 
 static void compute_hash_h3(const unsigned char *msg, uint32_t msg_len, unsigned char *hash_value) {
     /* TODO: replace with hash-to-curve
     * H3(m): Implemented using hash_to_field from [HASH-TO-CURVE], Section 5.2 using L = 48,
     * expand_message_xmd with SHA-256, DST = contextString || "nonce", and prime modulus equal to Order(). */
-    unsigned char prefix[26 + 5];
-    memcpy(prefix, hash_context_prefix, 26);
-    memcpy(prefix + 26, "nonce", 5);
-    compute_hash_with_prefix(prefix, sizeof(prefix), msg, msg_len, hash_value);
+    secp256k1_sha256 sha;
+    secp256k1_sha256_initialize(&sha);
+    secp256k1_sha256_write(&sha, hash_context_prefix_h3, sizeof(hash_context_prefix_h3));
+    secp256k1_sha256_write(&sha, msg, msg_len);
+    secp256k1_sha256_finalize(&sha, hash_value);
 }
 
 static void compute_hash_h4(const unsigned char *msg, uint32_t msg_len, unsigned char *hash_value) {
     /* H4(m): Implemented by computing H(contextString || "msg" || m). */
```

Comment:
1. in this comment, we could replace `contextString || "msg"` with the value of `hash_context_prefix_h4`: `"FROST-secp256k1-SHA256-v11msg"`;
2. `m` could become `msg` (without quotes), since the name of the formal parameter is `msg` in the function;
3. for `compute_hash_h{1,2,3}()` the substitution is a bitter more difficult, since the comments seem to be taken from the specification, and that has a certain value on its own. Nonetheless, it would be good if the documentation was as much as possible aligned with the code. What to do?